### ISSUE TITLE
null/row in spikes bleeds over when converted to latents

### ISSUE THREAD

**Issue by mobeets** [mobeets/nullSpaceControl#308]
```
NB = D.blocks(2).fDecoder.NulM2;
RB = D.blocks(2).fDecoder.RowM2;
NBn = D.blocks(2).nDecoder.NulM2;
RBn = D.blocks(2).nDecoder.RowM2;
Y = D.blocks(2).spikes;
Z = D.blocks(2).latents;
Dc = D.simpleData.nullDecoder;

Y_noNull = Y*(NBn*NBn');
Z_noNull = tools.convertRawSpikesToRawLatents(Dc, Y_noNull');
max(abs(Z_noNull*NB)) % should be zeros

Y_noRow = Y*(RBn*RBn');
Z_noRow = tools.convertRawSpikesToRawLatents(Dc, Y_noRow');
max(abs(Z_noRow*NB)) % should be zeros
```

```

ans =

  Columns 1 through 6

    5.5560    7.0545    6.1932    3.6993    4.5101    3.1230

  Columns 7 through 8

    4.2214    4.4351


ans =

  Columns 1 through 6

    3.3699    4.8450    7.7420    2.8869    3.6231    1.4137

  Columns 7 through 8

    5.5919    5.4617
```

**Comment by mobeets** [comment]
need to just keep the same A, c, where baseline spikes in spike space results in zero row space, zero null space.

that way for spikes we can write:

Bu + c (so "c" contains the component contributed by the mean spike count in B)

where `B = B0 P` where P is whatever goes from spikes to latents via FA

**Comment by mobeets** [comment]
okay yes, this is true. the decoders for spikes and latents seem to be correct.

the problem I think is the mean-centering of spikes. the mean, mu, will contribute some amount of row space activity, and some amount of null space activity, so row and null space activity for spikes is always relative to this mu value.

I can also confirm that if you account for this, spike null space activity seems to be maintained when you map to latents.

but row space activity does not...

but the decoder should be wholly in the latent space, i.e., row space activity at the level of spikes should be contained within the latent conversion matrix, so this row space activity should be exactly the same for factors. not sure why it's not.

In other words, if M is the mapping from spikes to latents, and RB is the row space for spikes,

then `Y*(M*M')*RB` should be equal to `Y*RB`. but it's not.

**Comment by mobeets** [comment]
so shouldn't the row space for spikes just be `M*RBf`, where RBf is the row space for factors?

**Comment by mobeets** [comment]
okay, for baseline i am only fitting two points per thetaActualGrp16. the resulting null space values (top two dims) still show tuning, on average and just the points themselves.

![screen shot 2016-09-02 at 11 49 56 am](https://cloud.githubusercontent.com/assets/1677179/18209815/7ad1b01a-7103-11e6-93cc-877313429401.png)
![screen shot 2016-09-02 at 11 48 27 am](https://cloud.githubusercontent.com/assets/1677179/18209816/7ad21e60-7103-11e6-9958-5751dc7939e7.png)

so maybe it's possible to just look at these points individually. picking two, for example, and calculating the resulting spikes.

**Comment by mobeets** [comment]
as near as i can tell, this is because:

`rad2deg(subspace(NBn, sigmainv*beta'*NB)) % 88x86, 88x8`

is around 45 deg.

where NBn is the null space of spikes, NB is the null space of factors, and beta is the mapping from spikes to latents.

whereas `subspace(sigmainv*beta'*RB, RBn)` is 0.

**Comment by mobeets** [comment]
so if i do:

```
M = sigmainv*beta';
Y_noNull = Y*(RBn*RBn'); % Y*M*(RB*RB')*M'
Z_noNull = Y_noNull*M;
[u,s,v] = svd(Z_noNull)
```

Then `Z_noNull` is still 2d (i.e., diag(s) has only two non-zero entries), as desired, but then if you now look at `Z_noNull*NB` it has some variance, meaning NB intersects with the spike-level row space activity.

But then, if you now set the latent level row space to be those 2d vectors from svd on Z_noNull, then you end up having the same problem when projecting into null space...

**Comment by mobeets** [comment]
turns out this is because FA is not an orthogonal projection. it's a linear mapping, but not an orthogonal projection

https://en.wikipedia.org/wiki/Projection_(linear_algebra)#Orthogonal_projections

orthogonal projections have the form `A*(A'*A)^{-1}*A'`
